### PR TITLE
[#265] initial manual

### PR DESCRIPTION
**Pull request by GuChad369**
- Add a new initial `/manual` directory.
- Modify the `CMake` file to enable the use of `pandoc` and `LaTeX` for generating `.html` and `.pdf` files from `Markdown`.
- Update the Git `.workflows`.

### CODE REVIEW

**Comment by jesperpedersen** [comment]
```
-- System is Linux
-- pandoc found
CMake Error at cmake/FindPdflatex.cmake:14 (execute_process):
  execute_process failed command indexes:

    1: "Child return code: 1"

Call Stack (most recent call first):
  CMakeLists.txt:85 (find_package)
```

```
which pdflatex
/usr/bin/pdflatex
```

```
pdflatex --version
pdfTeX 3.141592653-2.6-1.40.25 (TeX Live 2023)
kpathsea version 6.3.5
Copyright 2023 Han The Thanh (pdfTeX) et al.
There is NO warranty.  Redistribution of this software is
covered by the terms of both the pdfTeX copyright and
the Lesser GNU General Public License.
For more information about these matters, see the file
named COPYING and the pdfTeX source.
Primary author of pdfTeX: Han The Thanh (pdfTeX) et al.
Compiled with libpng 1.6.37; using libpng 1.6.37
Compiled with zlib 1.2.13; using zlib 1.2.13
Compiled with xpdf version 4.04
```

**Comment by jesperpedersen** [comment]
You need to update `README.md` and `pgmoneta.spec` with the files

**Comment by jesperpedersen** [comment]
Make sure that you verify on Fedora / RHEL 8 / RHEL 9

**Comment by GuChad369** [comment]
Could you provide some guidance on the pgmoneta.spec file? I am not familiar with it.

**Comment by jesperpedersen** [comment]
You need to make

* A directory (see around https://github.com/pgmoneta/pgmoneta/blob/main/pgmoneta.spec#L33)
* Copy the .pdf and .html in (see around https://github.com/pgmoneta/pgmoneta/blob/main/pgmoneta.spec#L71)
* Define them (see around https://github.com/pgmoneta/pgmoneta/blob/main/pgmoneta.spec#L122)

Take a look at https://github.com/pgmoneta/pgmoneta/blob/main/doc/RPM.md

**Comment by GuChad369** [comment]
I've updated the README and `pgmoneta.spec` files; the RPM was successfully tested on Fedora.

**Comment by jesperpedersen** [comment]
cmake is ok, but

```
[ 98%] Generating HTML documentation
[100%] Generating PDF documentation
Error producing PDF.
! LaTeX Error: File `footnote.sty' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: sty)

Enter file name: 
! Emergency stop.
<read *> 
         
l.86 \makesavenoteenv

make[2]: *** [src/CMakeFiles/all_docs.dir/build.make:100: doc/pgmoneta-manual.en.pdf] Error 43
make[1]: *** [CMakeFiles/Makefile2:254: src/CMakeFiles/all_docs.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

**Comment by jesperpedersen** [comment]
```
dnf install -y texlive-latex-extra
Last metadata expiration check: 0:20:02 ago on Fri 19 Apr 2024 10:14:28 PM EDT.
No match for argument: texlive-latex-extra
Error: Unable to find a match: texlive-latex-extra
```

```
rpm -qa | grep texlive-scheme-basic
texlive-scheme-basic-svn54191-69.fc39.noarch
```

**Comment by jesperpedersen** [comment]
Its Fedora 39

**Comment by GuChad369** [comment]
We're facing a problem with `pdflatex` as it doesn't have a specific library. To use `pdflatex`, we must download additional libraries like `TeX Live`, but each system offers different libraries for LaTeX. Our project also requires the `footnote.sty` package and different libraries may not offer the same method to verify its availability. Should we focus on solving this issue only for the Fedora system?

**Comment by jesperpedersen** [comment]
The `README.md` should cover Fedora, the rest should be documented in https://github.com/pgmoneta/pgmoneta/blob/main/doc/DISTRIBUTIONS.md

**Comment by jesperpedersen** [comment]
The key package for Fedora is `texlive-mdwtools`

**Comment by GuChad369** [comment]
I used two other ways to install the related libraries. `texlive-mdwtools` does not include `pdflatex`.

**Comment by jesperpedersen** [comment]
Yeah, I know - but we need it for the style

**Comment by GuChad369** [comment]
Okay, let’s try it.

**Comment by GuChad369** [comment]
- I've restructured the manual without splitting it into two separate documents, as I noticed duplicate content for both users and developers. Separating them would result in excessive repetition. 
- We can enhance the content further at a later stage.

**Comment by jesperpedersen** [comment]
There are some formatting issues, especially in "Configuration" chapter.

And, we don't need "Architecture" under introduction that is an advanced developer chapter - so close to last.

Take a step back, and think about what a new user of pgmoneta would like to know about going from install to their first backup

File: doc/manual/00-head.en.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+% pgmoneta Manual
```
Comment (jesperpedersen) [inline doc/manual/00-head.en.md:1]: We don't need "Manual"

File: doc/manual/00-head.en.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+% PostgreSQL
```
Comment (jesperpedersen) [inline doc/manual/00-head.en.md:2]: Remove

File: doc/manual/00-head.en.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
+% April 18, 2024 (0.10.1)
```
Comment (jesperpedersen) [inline doc/manual/00-head.en.md:3]: Remove

File: doc/manual/01-introduction.en.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
+\newpage
+
+# Introduction
+
+[**pgmoneta**](https://github.com/pgmoneta/pgmoneta) is a backup / restore solution for [PostgreSQL](https://www.postgresql.org).
+
+**pgmoneta** is named after the Roman Goddess of Memory.
+
```
Comment (jesperpedersen) [inline doc/manual/01-introduction.en.md:8]: We need something something about RPO and RTO in the Introduction, but we could take that later

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by GuChad369** [comment]
I made some formatting changes. While `pandoc`'s default compilation can't handle all `Markdown` formatting, if we want a perfect HTML format, we must add a `CSS` file when `pandoc` compiles the HTML.

**Comment by jesperpedersen** [comment]
We can likely find ways to improve the layout, colors, ...

**Comment by jesperpedersen** [comment]
Rebased, and merged.

Thanks for your contribution !

**Pull request closed without merging**